Fix the initialization of amd::Monitor objects

choumin opened the pull request:
Since the name variable of amd::Monitor had been removed in the commit `f19260d568853a2a60cb7b8beb747e4eaab12dc1`, fix the initialization of amd::Monitor objects.

## Motivation

<!-- Explain the purpose of this PR and the goals it aims to achieve. -->

## Technical Details

<!-- Explain the changes along with any relevant GitHub links. -->

## Test Plan

<!-- Explain any relevant testing done to verify this PR. -->

## Test Result

<!-- Briefly summarize test outcomes. -->

## Submission Checklist

- [ ] Look over the contributing guidelines at https://github.com/ROCm/ROCm/blob/develop/CONTRIBUTING.md#pull-requests.

AidanBeltonS commented on projects/clr/rocclr/device/device.cpp:
```diff
@@ -87,7 +87,7 @@ static_assert(static_cast<uint32_t>(device::Memory::MemAccess::kMemAccessReadWri
 namespace amd {
 
-amd::Monitor Device::lockP2P_("Lock P2P ON/OFF");
+amd::Monitor Device::lockP2P_{};
```
Does this change accidentally alter the monitors behavior? The amd::Monitor class accepts a bool to turn it into a recursive monitor. I suspect that the string is being implicitly converted into a bool and taking that path. So this may need to be `amd::Monitor Device::lockP2P_(true);` if we want this to not change the monitor type. Same point for other code change

choumin replied:
Yes, no changing the monitor type is more safe. I have fixed it. Thanks.

choumin committed:
Fix the initialization of amd::Monitor objects

Since the name variable of amd::Monitor had been removed
in the commit `f19260d568853a2a60cb7b8beb747e4eaab12dc1`,
fix the initialization of amd::Monitor objects.

Signed-off-by: Min Zhou <zhoumin@bupt.cn>

AidanBeltonS reviewed (approved):
Comment addressed. Should be good to merge if it passes CI

choumin committed:
Merge branch 'develop' into fix_monitor
